packages/opencensus-instrumentation-redis/test/test-redis.ts

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CoreTracer, logger, RootSpan, Span, SpanEventListener, SpanKind} from '@opencensus/core';
+import * as assert from 'assert';
+import * as redis from 'redis';
+
+import {plugin} from '../src';
+
+/** Collects ended root spans to allow for later analysis. */
+class RootSpanVerifier implements SpanEventListener {
+  endedRootSpans: RootSpan[] = [];
+
+  onStartSpan(span: RootSpan): void {
+    return;
+  }
+  onEndSpan(root: RootSpan) {
+    this.endedRootSpans.push(root);
+  }
+}
+
+/**
+ * Asserts root spans attributes.
+ * @param rootSpanVerifier An instance of rootSpanVerifier to analyse RootSpan
+ * instances from.
+ * @param expectedName The expected name of the first root span.
+ * @param expectedKind The expected kind of the first root span.
+ */
+function assertSpan(
+    rootSpanVerifier: RootSpanVerifier, expectedName: string,
+    expectedKind: SpanKind, verifyAttribute?: (span: Span) => boolean) {
+  assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 1);
+  assert.strictEqual(rootSpanVerifier.endedRootSpans[0].spans.length, 1);
+  assert.strictEqual(
+      rootSpanVerifier.endedRootSpans[0].spans[0].name, expectedName);
+  assert.strictEqual(
+      rootSpanVerifier.endedRootSpans[0].spans[0].kind, expectedKind);
+  if (typeof verifyAttribute === 'function') {
+    for (const span of rootSpanVerifier.endedRootSpans[0].spans) {
+      assert(verifyAttribute(span), 'failed to verify attribute');
+    }
+  }
+}
+
+describe('RedisPlugin', () => {
+  // For these tests, mongo must be runing. Add OPENCENSUS_REDIS_TESTS to run
+  // these tests.
+  const OPENCENSUS_REDIS_TESTS = process.env.OPENCENSUS_REDIS_TESTS as string;
+  const OPENCENSUS_REDIS_HOST = process.env.OPENCENSUS_REDIS_HOST as string;
+  let shouldTest = true;
```

Comment:
Could this be switched to `const` and just be `const shouldTest = process.env.OPENCENSUS_REDIS_TESTS`? (That would allow removing the separate `OPENCENSUS_REDIS_TESTS` variable as well)

Then there could be
```
if (!shouldTest) {
    console.log('Skipping test-redis. Run REDIS to test');
}
```

Reply:
We can't use a `const` there because if we fail to connect to the redis instance later on we skip all the tests by using the same variable.